cinder/base.html

```diff
@@ -92,46 +92,24 @@
         {% endif %}
         {% endblock %}
     </footer>
-
+    
+    {%- block scripts %}
     <script src="//ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
     <script src="{{ 'js/bootstrap-3.0.3.min.js'|url }}"></script>
     <script src="{{ 'js/highlight.pack.js'|url }}"></script>
     <script>hljs.initHighlightingOnLoad();</script>
-    <script>
-    var base_url = '{{ base_url }}';
-    </script>
+    <script>var base_url = {{ base_url | tojson }}</script>
+    {% if config.shortcuts %}
+        <script>var shortcuts = {{ config.shortcuts | tojson }}</script>
+    {% endif %}
     <script src="{{ 'js/base.js'|url }}"></script>
     {%- for path in config['extra_javascript'] %}
     <script src="{{ path|url }}"></script>
     {%- endfor %}
+    {%- endblock %}
 
-    <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
-        <div class="modal-dialog">
-            <div class="modal-content">
-                <div class="modal-header">
-                    <button type="button" class="close" data-dismiss="modal">
-                        <span aria-hidden="true">&times;</span>
-                        <span class="sr-only">Close</span>
-                    </button>
-                    <h4 class="modal-title" id="exampleModalLabel">Search</h4>
-                </div>
-                <div class="modal-body">
-                    <p>
-                        From here you can search these documents. Enter your search terms below.
-                    </p>
-                    <form role="form">
-                        <div class="form-group">
-                            <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
-                        </div>
-                    </form>
-                    <div id="mkdocs-search-results"></div>
-                </div>
-                <div class="modal-footer">
-                </div>
-            </div>
-        </div>
-    </div>
-
+    {% if 'search' in config['plugins'] %}{%- include "search-modal.html" %}{% endif %}
```

Comment:
Does this mean that users need to include an explicit setting in the yml file to include the search modal support in their build?

Reply:
This was copied from the [base.html in mkdocs](https://github.com/mkdocs/mkdocs/blob/master/mkdocs/themes/mkdocs/base.html#L203).

According to the mkdocs...docs about plugin settings -> https://www.mkdocs.org/user-guide/configuration/#plugins

> If the plugins config setting is defined in the mkdocs.yml config file, then any defaults (such as search) are ignored and you need to explicitly re-enable the defaults if you would like to continue using them:

So, I'm assuming that if someone implements their custom .yml file with the `plugins` config set, they need to be sure to explicitly list out `search` otherwise it will be disabled. If they do not have any plugins setting in their .yml then search will be enabled.

Reply:
Great.  That is helpful.  Thanks Tim.